### ISSUE TITLE
Responsiveness

### ISSUE THREAD

**Issue by vanhoutenbos** [select2/select2-bootstrap-theme#30]
Hello,

I thought bootstrap is created to make websites responsive, but this plugin doesn't fix the responsiveness issues at all.
It does style it to a nice bootstrap looking button but without the responsiveness its useless.

Could this be implemented.

**Comment by psohm** [comment]
well it is strange.
when we change the size of the window, it doesn't readapt
but if we refresh the page it has the good size ..

**Comment by Jean-PaulvHouten** [comment]
Exactly, its kinda strange, since there is code made for responsiveness, but it doesn't seem to work dynamicly. (I am logged in on my work account now, same guy as who asked the question though)

**Comment by rtsalamone** [comment]
Quick Hack that may help someone else until this is fixed:
<code><pre>
    $(function(){
        setTimeout(function() {
            $('span.select2-container.select2-container--bootstrap').width('100%');
            }, 1000);
        });
</pre></code>
Since there doesn't seem to be an "initialized" event to hook into, I'm waiting one second after doc ready to set the spans with to 100%, which makes them responsive.

 As I said, it's a total hack, but it will let you at least demo the thing!

**Comment by ecmel** [comment]
If I construct the select2 as per the examples, it works out of the box:

```
$.fn.select2.defaults.set("theme", "bootstrap");
$("#single").select2({
  width: null
});
```

Is this bug still valid?

**Comment by klarkc** [comment]
The bug still occurs, the @ecmel workaround works

Another way, to disables the width parameter by default:

``` php
$.fn.select2.defaults.set("width", null);
```

**Comment by onebree** [comment]
I am using select2 4.0.3, and select2-bootstrap 0.1.0-beta.6

Can confirm that this bug still exists.

**Comment by nunesvictor** [comment]
this workaround fixes the issue:

```
// select2.init.js
$.fn.select2.defaults.set("theme", "bootstrap");
$.fn.select2.defaults.set("width", null);

$(document).ready(function () {
    $('select').select2();
});
```

using: select2 4.0.1 and select2-bootstrap 0.1.0-beta.10

**Comment by larigyn** [comment]
sir, 
It overlaps.
![p1](https://user-images.githubusercontent.com/26628446/27833712-82a333c4-6106-11e7-9f34-01686c7541a5.png)

![p2](https://user-images.githubusercontent.com/26628446/27833716-878f607e-6106-11e7-992a-532477a6dc74.png)

**Comment by ale-diez** [comment]
[Here's](https://github.com/select2/select2/issues/3278#issuecomment-336092563) what fixes it for me.
Basically, just reset the select's parent container width after initializing it

```
$(function() { 
  $('#controlID').select2({ theme: 'bootstrap' }); 
  $('#controlID').parent().find('.select2-container').css('width', '');  <-- This part here
});
```

The theme shouldn't give a fixed with on this element though
